builtin/stash--helper.c

```diff
@@ -689,6 +695,50 @@ static int show_stash(int argc, const char **argv, const char *prefix)
 	return ret;
 }
 
+static int do_store_stash(const char *stash_msg, const char *w_commit)
+{
+	struct object_id obj;
+	int ret = 0;
+
+	ret = get_oid(w_commit, &obj);
+	if (!ret) {
+		ret = update_ref(stash_msg, ref_stash, &obj, NULL,
+			REF_FORCE_CREATE_REFLOG, 0);
+	}
+
+	if (ret && !quiet)
+		fprintf_ln(stderr, _("Cannot update %s with %s"), ref_stash, w_commit);
+
+	return ret;
+}
```

Comment:
I do not see any reason for this code to live in its own function. Why not just fold it into `store_stash()`? It'll be easier to read, too, as it does not disrupt the flow (you have to read `store_stash()` first, then scroll up to read `do_store_stash()` right now...)

Reply:
Right.